encoding/encoding.go

```diff
@@ -18,9 +18,28 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package tsz
+package encoding
 
-func leadingAndTrailingZeros(v uint64) (int, int) {
+// Bit is just a byte
+type Bit byte
+
+// NumSig returns the number of significant values in a uint64
+func NumSig(v uint64) uint8 {
+	if v == 0 {
+		return 0
+	}
+
+	numLeading := uint8(0)
```

Comment:
nit: can pull out the logic to calculate leading zeros into a separate function and use it both here and `LeadingAndTrailingZeros`